_posts/2020-02-03-ftp.md

```diff
@@ -0,0 +1,80 @@
+---
+layout: post
+current: post
+cover: FTP.jpg
+title: Why you shouldn't be using FTP in 2020
+date: 2020-02-01 17:00:00
+category: tutorials
+author: Fiona McAllister
+tags: [open-source]
+class: post-template
+---
+
+## Introduction
+
+For many services, it is not enough to establish a connection. There are instances where a developer might want to transfer files from one server to another. FTP (File Transfer Protocol) is one of many available file transfer protocols that are available to use for servers. It enables the transfer of files from one machine or server to another over the internet. FTP itself is just an extension of the TCP protocol and uses the same core principles to establish and maintain a connection. The problem with this is that FTP (and TCP) are now almost 50 years old. 
+
+Since the development of FTP and TCP in the 1970s, cybersecurity has become a growing concern. FTP does not encrypt data and by default, neither does TCP. This means that even if you use a username and password over FTP, your data can still be compromised. Here we'll underline why exactly FTP is insecure and alternatives that you can use to keep your data, especially your credentials, secure.
+
+**Reading time: 5 min**
+
+## What is FTP and why is it insecure?
+
+FTP is a system that allows for the transfer of data and files between two systems, a server and a client. The server acts as a data store that the client can access and exchange files between and usually uses port 21. It was developed in the 1970s by Abhay Buhshan as an extension to the TCP/IP protocol in order to transfer files. At the time, it was revolutionary, allowing a user to remotely access data. However, this was back when computers were simpler and text-based and it didn't take into account security, at least not from a hacker's perspective. 
+
+FTP now is a fairly redundant mode of file transfer. It is inefficient, and although it has been updated somewhat over the years, it is still an old protocol designed for non-graphical systems. The problem lies in the fact that many applications still use FTP today, despite support being phased out by various distributors including the [Debian Project](https://www.debian.org/News/2017/20170425). 
+
+But why is it insecure? On the surface, it looks as though FTP can be secured through username and password. However, this isn't really the case, or rather it's not strictly speaking secure. This is in part because FTP doesn't actually *encrypt* its data. Any authentication data is transferred as a plaintext file making it particularly vulnerable to spoofing. In other words, an attacker could 'listen' to your packets, gaining information regarding your login credentials. This is because of the plaintext format, your username and password can be easily read and used to breach your system. Beyond this as well is the fact that FTP is no longer maintained, it has not actually been updated to be compatible with the modern world.
```

Comment:
* "vulnerable to spoofing" -> "vulnerable to sniffing"
* "beyond this as well is the fact that FTP is no longer maintained" - I'm pretty sure the popular FTP servers (such as vsftpd) are still maintained. The problem is that the [FTP RFC](https://tools.ietf.org/html/rfc959) didn't really take security into consideration, as it's from 1985.

Reply:
isn't vsftpd a FTP(S) server though?

Reply:
vsftpd supports both. By default SSL/TLS is disabled.

http://vsftpd.beasts.org/vsftpd_conf.html